proposals/lip-replace-payloadhash-with-merkle-tree-root-in-block-header.md

```diff
@@ -0,0 +1,73 @@
+
+
+### Block Verification
+
+To verify the validity of a block, first the Merkle root is calculated from the transactions included in the block payload and checked against the `transactionRoot` stored in the block header. The same set of transactions sorted in a different way would result in a different Merkle root, and consequently an invalid block. In this sense, the `transactionRoot` contains information about the order in which the transactions were applied to the blockchain. Then, the transactions are verified and applied in the same order in which they have been inserted in the payload. 
+
+
+### Transactions Ordering
+
+To reproduce the same Merkle root consistently, the transactions have to be sorted in a consistent way. We consider two viable options:
+
+1. Sort by transaction ID: transactions are sorted according to their transaction ID in ascending order. Using this sorting method, it is easy to give a proof-of-absence for a certain transaction by giving a proof-of-inclusion for two transactions next to each other in the tree and whose IDs bound the target transaction ID. 
+2. Sort by application order: after [LIP 0026][establish-validity] comes into effect, transactions will be applied to the blockchain state in a specific order, chosen by the forger. Therefore, transactions will be already sorted in this way in the block payload. Using this sorting method to build the Merkle tree and consequently calculate the `transactionRoot` allows to implicitly make the application order an immutable part of the block header. This approach does not allow for an efficient proof-of-absence, however this is only a marginal downside as proofs-of-absence do not have relevant use-case scenarios in the current protocol.
+
+As mentioned above, we choose option 2.
+
+
+## Specification
+
+We assume [LIP 0019 "Use full SHA-256 hash of transaction as transactionID"](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0019.md) is already effective, and transaction IDs are 32 bytes long. 
+
+Given a Merkle tree and corresponding Merkle root `merkleRoot` built from the array `transactionIDsArray` according to the specifications defined in [LIP 0031][introduce-MT], the `transactionRoot` is set as `transactionRoot = merkleRoot`. Here `transactionIDsArray` is an array of `bytes` containing the serialized transaction ids. The `transactionRoot` is then stored in the block header instead of `payloadHash`.
```

Comment:
```suggestion
Given a Merkle tree and corresponding Merkle root `merkleRoot` built from the array `transactionIDsArray` according to the specifications defined in [LIP 0031][introduce-MT], the `transactionRoot` is set as `transactionRoot = merkleRoot`. Here `transactionIDsArray` is an array of `bytes` containing the serialized transaction IDs. The `transactionRoot` is then stored in the block header instead of `payloadHash`.
```